scipy/_lib/messagestream.pyx

```diff
@@ -88,3 +88,15 @@ cdef class MessageStream:
         if not self._removed:
             stdio.remove(self._filename)
             self._removed = 1
+
+    def __dealloc__(self):
```

Comment:
Can't you just rename `__del__` to `__dealloc__`? You might also have to change `def close` to `cpdef close`.

Reply:
> Can't you just rename `__del__` to `__dealloc__`?

Yes, in fact, we should do this. From the [documentation](https://cython.readthedocs.io/en/latest/src/userguide/special_methods.html#finalization-method-dealloc):

> Note: There is no \_\_del\_\_() method for extension types

> You might also have to change `def close` to `cpdef close`

That sounds right.

Reply:
Done, thanks!